### PR TITLE
Parity: Handle receipt of unmined transactions

### PR DESCRIPTION
**Pull request by hackaugusto**
# Description

Parity will always return a receipt, even if the transaction is not
mined:

https://wiki.parity.io/JSONRPC-eth-module#eth_gettransactionreceipt

This means that if a transaction is not mined, there will be a receipt
with a value of `None` for the `blockNumber`. And an exception is raised
when that value is added to the confirmation block.

## PR review check list

Quality check list that cannot be automatically verified.

- Safety
    - [x] The changes respect the necessary conditions for safety (https://raiden-network-specification.readthedocs.io/en/latest/smart_contracts.html#protocol-values-constraints)
-  Code quality
    - [x] Error conditions are handled
    - [x] Exceptions are propagated to the correct parent greenlet
    - [x] Exceptions are correctly classified as recoverable or unrecoverable
- Compatibility
    - [x] State changes are forward compatible
    - [x] Transport messages are backwards and forward compatible
- Commits
    - [x] Have good messages
    - [x] Squashed unecessary commits
- If it's a bug fix:
    - Regression test for the bug
        - [ ] Properly covers the bug
        - [ ] If an integration test is used, it could not be written as a unit test
- Documentation
    - [x] A new CLI flag was introduced, is there documentation that explains usage?
- Specs
    - [x] If this is a protocol change, are the specs updated accordingly? If so, please link PR from the specs repo.
- Is it a user facing feature/bug fix?
    - [ ] Changelog entry has been added

### CODE REVIEW

File: raiden/network/rpc/client.py [hunk 1]
```diff
@@ -651,20 +662,39 @@ def poll(self, transaction_hash: TransactionHash) -> Dict[str, Any]:
...
+            #   to: None
+            #   transactionHash: The transaction's hash
+            #   transactionIndex: 0
+            #
+            # Geth only returns a receipt if the transaction was mined on the
+            # canonical chain. https://github.com/raiden-network/raiden/issues/4529
+            is_transaction_mined = tx_receipt and tx_receipt.get("blockNumber") is not None
+
+            if is_transaction_mined:
                 confirmation_block = (
```
Comment (hackaugusto) [inline raiden/network/rpc/client.py]: Without the check `tx_receipt.get("blockNumber") is not None` above, here `tx_receipt["blockNumber"] + self.default_block_num_confirmations` we would have something like `None + 5` which raises an exception.

**Comment by pirapira** [comment]
Does this fix a user-facing bug?  Then consider adding a Changelog entry.

Shall we test this problem?

**Comment by hackaugusto** [comment]
> Shall we test this problem?

It is tested by https://github.com/raiden-network/raiden/pull/4782 (indirectly, that is how I found the problem)

**Comment by pirapira** [comment]
Then no extra test is necessary.  I think a Changelog entry is good for affected Parity users.

**Pull request merged**